corehq/apps/data_analytics/tests/test_malt_generator.py

```diff
@@ -279,70 +273,51 @@ def setUp(self):
         self.monthspan = DateSpan.from_month(1, 2022)
         self.app_row = create_mock_app_row_for_malt_tests()
         self.user = create_mock_user_for_malt_tests()
-        self.app_data = MaltAppData(
-            AMPLIFIES_NOT_SET,
-            AMPLIFIES_NOT_SET,
-            DEFAULT_MINIMUM_USE_THRESHOLD,
-            DEFAULT_EXPERIENCED_THRESHOLD,
-            False
-        )
+        self.app_data = create_malt_app_data()
+
+        self.patcher = patch('corehq.apps.data_analytics.malt_generator._get_malt_app_data')
+        self.mock_get_malt_app_data = self.patcher.start()
+        self.mock_get_malt_app_data.return_value = self.app_data
+
+    def tearDown(self):
+        self.patcher.stop()
+        super().tearDown()
```

Comment:
Take a look at the bottom example for patching [here](https://docs.python.org/3/library/unittest.mock.html#patch-methods-start-and-stop). It's less error-prone (and allows you to eliminate teardown entirely) if you use self.addCleanup(self.patcher.stop). Because the patcher is only used within the setup method, as well, it allows you to just refer to it as patcher, rather than self.patcher

Reply:
See 80a39f63f862adc8bdea4bfc8cc3e1f3889082e1